ophyd/signal.py

```diff
@@ -1711,6 +1711,9 @@ def set(self, value, *, timeout=None, settle_time=None):
         --------
         Signal.set
         '''
+        if timeout is None:
+            timeout = self.timeout
```

Comment:
In `put` we use `self.write_timeout` instead of `self.timeout`, we should probably be consistent with that here.

The other "fix" may be to change the default value of timeout to be `timeout=DEFAULT_WRITE_TIMEOUT` like put (as I assume it eventually falls through to that).

Reply:
When I check my test case, `self.write_timeout` is `None` while `self.timeout` is as defined (by me) via `EpicsSignalBase.set_default_timeout(timeout=10, connection_timeout=5)`.

So we need a second test for `None`ness.

Reply:
```
In [6]: s.write_timeout is None                                                                                                                    
Out[6]: True

In [7]: s.timeout                                                                                                                                  
Out[7]: 10
```